src/components/UpcomingEventsCard/index.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import './style.css';
+import bg2 from './img/bg_2.png';
+import { charityAPI } from '../../clients';
+
+class UpcomingEventsCard extends React.Component {
+  state = {
+    date: {},
+    loading: true,
+    error: false,
+    errorMessage: ''
+  };
+
+  _getData = () => {
+    this.setState({ loading: true });
+    charityAPI('/featured-cause')
+      .then(({ data }) => {
+        this.setState({
+          data: data,
+          loading: false,
+          error: false,
+          errorMessage: ''
+        });
+      })
+      .catch(error => {
+        this.setState({
+          loading: false,
+          error: true,
+          errorMessage: "couldn't fetch data"
+        });
+      });
+  };
+
+  componentDidMount() {
+    this._getData();
```

Comment:
I liked how you separated your data fetching in a separate function and cleaned up your react lifecycle method.